benches/cctv.rs

```diff
@@ -0,0 +1,89 @@
+use std::io::BufRead;
+
+use criterion::{criterion_group, criterion_main, BatchSize, Criterion};
+use crypto_bigint::U1024;
+use rand_chacha::ChaCha8Rng;
+
+use crypto_primes::{is_prime_with_rng, sieve_and_find, SieveFactory};
+use rand_core::SeedableRng;
+
+struct HardcodedSieve {
+    candidates: Vec<String>,
+    idx: usize,
+}
+impl HardcodedSieve {
+    fn new() -> Self {
+        Self {
+            candidates: std::fs::read("./benches/rsa.bench.2048.txt")
+                .expect("file present")
+                .lines()
+                .map(|l| l.unwrap().to_owned())
+                .collect(),
+            idx: 0,
+        }
+    }
+}
+impl Iterator for HardcodedSieve {
+    type Item = U1024;
+    fn next(&mut self) -> Option<Self::Item> {
+        self.idx += 1;
+        self.candidates.get(self.idx).map(|str| U1024::from_be_hex(str))
+    }
+}
+
+impl SieveFactory for HardcodedSieve {
+    type Item = U1024;
+    type Sieve = Self;
+    fn make_sieve(
+        &mut self,
+        _rng: &mut impl rand_core::CryptoRngCore,
+        _previous_sieve: Option<&Self::Sieve>,
+    ) -> Option<Self::Sieve> {
+        Some(Self {
+            candidates: self.candidates.clone(),
+            idx: 0,
+        })
+    }
+}
+/// CCTV stands for Community Cryptography Test Vectors[1]. This benchmark uses the
+/// "rsa.bench.2048.txt" test vector, which is a file of 708 1024-bit long candidates for prime
```

Comment:
I wonder why it has "2048" in its name if the numbers are 1024-bit

Reply:
Same! I was convinced it was 2048 bit numbers but then I realized they came to it from implementing RSA keygen so multiplying two 1024 numbers to build one 2048 bit key...